packages/eas-cli/src/commands/env/delete.ts

```diff
@@ -0,0 +1,110 @@
+import { Flags } from '@oclif/core';
+
+import EasCommand from '../../commandUtils/EasCommand';
+import {
+  EASEnvironmentFlag,
+  EASNonInteractiveFlag,
+  EASVariableScopeFlag,
+  EASVariableSensitiveFlag,
+} from '../../commandUtils/flags';
+import { EnvironmentVariableScope } from '../../graphql/generated';
+import { EnvironmentVariableMutation } from '../../graphql/mutations/EnvironmentVariableMutation';
+import { EnvironmentVariablesQuery } from '../../graphql/queries/EnvironmentVariablesQuery';
+import Log from '../../log';
+import { promptAsync, toggleConfirmAsync } from '../../prompts';
+import { promptVariableEnvironmentAsync } from '../../utils/prompts';
+
+export default class EnvironmentVariableDelete extends EasCommand {
+  static override description = 'delete an environment variable by name';
+
+  static override flags = {
+    name: Flags.string({
+      description: 'Name of the variable to delete',
+    }),
+    ...EASVariableSensitiveFlag,
+    ...EASVariableScopeFlag,
+    ...EASEnvironmentFlag,
+    ...EASNonInteractiveFlag,
+  };
+
+  static override contextDefinition = {
+    ...this.ContextOptions.ProjectConfig,
+    ...this.ContextOptions.LoggedIn,
+  };
+
+  async runAsync(): Promise<void> {
+    let {
+      flags: { name, environment, 'non-interactive': nonInteractive, scope },
+    } = await this.parse(EnvironmentVariableDelete);
+    const {
+      privateProjectConfig: { projectId },
+      loggedIn: { graphqlClient },
+    } = await this.getContextAsync(EnvironmentVariableDelete, {
+      nonInteractive,
+    });
+    let variable;
+    let variables;
```

Comment:
Using `let`s is generally an antipattern so we should avoid using them if possible.

I believe that if move logic to query variables and select a variable to a function you should be able to do something like this:
```suggestion
    const variables = getVariablesForScopeAndEnvironmentAsync({ scope, environment });
    const variable = selectVariableToBeDeleted({ variables });
```